tests/datadriven/test_references.py

```diff
@@ -31,23 +32,89 @@ class ReferenceSetTest(datadriven.DataDrivenTest):
     a reference set, and verifies that it is consistent with the model
     built by the references.ReferenceSet object.
     """
-    def __init__(self, referenceSetId, baseDir):
-        super(ReferenceSetTest, self).__init__(referenceSetId, baseDir)
-        # Read in all the FASTA files in dataDir. Each reference within
-        # the reference set maps to a single FASTA file.
-        self._idFastaFileMap = {}
-        for path in glob.glob(os.path.join(self._dataDir, "*.fa.gz")):
-            filename = os.path.split(path)[1]
+    class ReferenceInfo(object):
+        """
+        Container class for information about a reference
+        """
+        def __init__(self, referenceSetId, fastaFileName):
+            filename = os.path.split(fastaFileName)[1]
             referenceId = "{}:{}".format(
                 referenceSetId, filename.split(".")[0])
-            self._idFastaFileMap[referenceId] = pysam.FastaFile(path)
+            self.id = referenceId
+            self.fastaFile = pysam.FastaFile(fastaFileName)
+            self.bases = self.fastaFile.fetch(self.fastaFile.references[0])
```

Comment:
Oops, sorry! My bad, didn't realise this was the test path. This brings up another issue: shouldn't we be using some other library to access the FASTA data (i.e. not pysam) as a sanity check?